File: Basics_with_JavaScript/56 Conditional Statements/More Exersises/03. Harvest.js
function vladi(input) {
  let lozeArea = Number(input[0]);
  let grapesPerSqm = Number(input[1]);
  let wineNeeds = Number(input[2]);
  let workersQuantity = Number(input[3]);
  let lozeForWineProduction = lozeArea * 0.4;

  let wineProducedL = (lozeForWineProduction * grapesPerSqm) / 2.5;

  difference1 = Math.floor(Math.abs(wineNeeds - wineProducedL));
  difference2 = Math.ceil(Math.abs(wineNeeds - wineProducedL));
  let workerWine = Math.ceil(difference2 / workersQuantity);

  if (wineProducedL < wineNeeds) {
    console.log(
      `It will be a tough winter! More ${difference1} liters wine needed.`
    );
  } else {
    console.log(`Good harvest this year! Total wine: ${wineProducedL} liters.`);
    console.log(
      `${difference2} liters left -> ${workerWine} liters per person.`
    );
  }
}

vladi([650, 2, 175, 3]);
vladi([1020, 1.5, 425, 4]);

File: Basics_with_Python/PythonBasicCourse/2_Conditional_Statements/6_Conditional_Statements_Excercises___Exercise/more_exercises/03._Harvest.py
import math
vineyard_area = int(input())
grapes_per_sqm_kg = float(input())
required_wine = int(input())
workers = int(input())

total_grape_production = vineyard_area * grapes_per_sqm_kg
grape_for_wine_kg = total_grape_production * 0.4
total_wine = grape_for_wine_kg / 2.5
ss = 0
if total_wine < required_wine:
    print(f"It will be a tough winter! More {int(math.floor(required_wine - total_wine))} liters wine needed.")
else:
    print(f"Good harvest this year! Total wine: {int(math.floor(total_wine))} liters.")
    ss = (total_wine - required_wine) / workers
    print(f"{int(math.ceil(total_wine - required_wine))} liters left -> {int(math.ceil(ss))} liters per person.")
